tests/console/verify_efi_mok.pm

```diff
@@ -0,0 +1,272 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Check EFI boot in images or after OS installation
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use utils qw(zypper_call is_efi_boot);
+use version_utils qw(is_leap is_opensuse is_sle);
+use Utils::Backends qw(is_hyperv is_svirt_except_s390x);
+use jeos qw(reboot_image set_grub_gfxmode);
+use constant {
+    SYSFS_EFI_BITS      => '/sys/firmware/efi/fw_platform_size',
+    FSTAB               => '/etc/fstab',
+    GRUB_DEFAULT        => '/etc/default/grub',
+    GRUB_CFG            => '/boot/grub2/grub.cfg',
+    SYSCONFIG_BOOTLADER => '/etc/sysconfig/bootloader'
+};
+
+sub get_expected_efi_settings {
+    my $settings      = {};
+    my $efi_rec_label = (is_opensuse() ? (lc get_var('DISTRI')) : 'sles');
+    my $efi_exec      = $efi_rec_label . '/grubx64.efi';
+
+    if (!get_var('DISABLE_SECUREBOOT', 0)) {
+        $efi_exec = $efi_rec_label . '/shim.efi';
+        $efi_rec_label .= '-secureboot';
+    }
+    $settings->{exec}  = '/EFI/' . $efi_exec;
+    $settings->{label} = $efi_rec_label;
+
+    return $settings;
+}
+sub efibootmgr_current_boot {
+    my $ebm_raw = script_output 'efibootmgr --verbose';
+    my $h       = {};
+
+    ($h->{bootid}) = $ebm_raw =~ /^BootCurrent:\s+(\d+)/m;
+    die("Missing BootCurrent: in efibootmgr output") unless (defined($h->{bootid}));
+
+    if ($ebm_raw =~ /^(Boot$h->{bootid}\*\s+(.+)\s+(\S+\([^\)]+\)\/?)+.*|Boot$h->{bootid}\*\s+(.+))$/m) {
+        $h->{label} = $2 // $4;
+    }
+    die("Missing lable in Boot$h->{bootid} entry") unless (exists($h->{label}) or defined($h->{label}));
```

Comment:
same here

Reply:
ah, my mistake. In this case you can check exists :)

Reply:
Not sure, if the condition fails, I think key `label` won't even exist. But I have typo there :D